pkg/cmd/server/api/validation/master.go

```diff
@@ -16,26 +16,60 @@ import (
 	"github.com/openshift/origin/pkg/util/labelselector"
 )
 
-func ValidateMasterConfig(config *api.MasterConfig) fielderrors.ValidationErrorList {
-	allErrs := fielderrors.ValidationErrorList{}
+type ValidationResults struct {
+	Warnings fielderrors.ValidationErrorList
+	Errors   fielderrors.ValidationErrorList
+}
+
+func (r *ValidationResults) AppendErrors(warnings fielderrors.ValidationErrorList, errors fielderrors.ValidationErrorList) {
```

Comment:
why does `AppendErrors` take `warnings` and `errors` and append both? not sure why this is necessary